components/clm/src/main/subgridWeightsMod.F90

```diff
@@ -580,52 +592,86 @@ subroutine check_weights (bounds, active_only)
           sumwtcol(c) = sumwtcol(c) + veg_pp%wtcol(p)
           sumwtlunit(l) = sumwtlunit(l) + veg_pp%wtlunit(p)
+          !topo_active_only = top_pp%active(t)
+          !if(topo_active_only) then  !TKT calculate only for active topounits
+          sumwttunit(t) = sumwttunit(t) + veg_pp%wttopounit(p)
           sumwtgcell(g) = sumwtgcell(g) + veg_pp%wtgcell(p)
+          !end if
        end if
     end do
 
     do c = bounds%begc,bounds%endc
-       if (.not. weights_okay(sumwtcol(c), active_only, col_pp%active(c))) then
-          write(iulog,*) trim(subname),' ERROR: at c = ',c,'total PFT weight is ',sumwtcol(c), &
+       tu = col_pp%topounit(c)
+       topo_active_only = top_pp%active(tu) 
+      ! write(iulog,*) trim(subname),' At c = ',c,'is topounit active = ', topo_active_only
+       if (topo_active_only) then ! Check only for the valid topounits
+       !   write(iulog,*) trim(subname),' At c = ',c,'total PFT weight is ',sumwtcol(c)
+          if (.not. weights_okay(sumwtcol(c), active_only, col_pp%active(c))) then
+             write(iulog,*) trim(subname),' ERROR: at c = ',c,'total PFT weight is ',sumwtcol(c), &
                          'active_only = ', active_only
-          error_found = .true.
+             error_found = .true.
+          end if
        end if
     end do
 
     do l = bounds%begl,bounds%endl
-       if (.not. weights_okay(sumwtlunit(l), active_only, lun_pp%active(l))) then
-          write(iulog,*) trim(subname),' ERROR: at l = ',l,'total PFT weight is ',sumwtlunit(l), &
+       tu = lun_pp%topounit(l)
+       topo_active_only = top_pp%active(tu) 
+      ! write(iulog,*) trim(subname),' At l = ',l,'is topounit active = ', topo_active_only
+       if (topo_active_only) then 
+        !  write(iulog,*) trim(subname),' At l = ',l,'total PFT weight is ',sumwtlunit(l)
+          if (.not. weights_okay(sumwtlunit(l), active_only, lun_pp%active(l))) then
+             write(iulog,*) trim(subname),' ERROR: at l = ',l,'total PFT weight is ',sumwtlunit(l), &
                          'active_only = ', active_only
-          error_found = .true.
+             error_found = .true.
+          end if
+       end if
+    end do
+    
+    do t = bounds%begt,bounds%endt       
+       topo_active_only = top_pp%active(t)
+     !  write(iulog,*) trim(subname),' At t = ',t,'is topounit active = ', topo_active_only
+       if (topo_active_only) then 
+       !   write(iulog,*) trim(subname),' At t = ',t,'total PFT weight is ',sumwttunit(t)
+          if (.not. weights_okay(sumwttunit(t), active_only, top_pp%active(t))) then
+             write(iulog,*) trim(subname),' ERROR: at t = ',t,'total PFT weight is ',sumwttunit(t), &
+                         'active_only = ', active_only
+             error_found = .true.
+          end if
        end if
     end do
 
     do g = bounds%begg,bounds%endg
+     !  write(iulog,*) trim(subname),' At g = ',g,'total PFT weight is ',sumwtgcell(g)
```

Comment:
Remove the commented out code

Reply:
Thanks! Commented out code has been removed.